### ISSUE TITLE
Apply bans / unbans in large rooms first

### ISSUE THREAD

**Issue by grahamc** [matrix-org/mjolnir#141]
**Is your feature request related to a problem? Please describe.**

I deploy Mjolnir in to just over 100 rooms. Applying (un)bans can take considerable amount of time: sometimes several minutes. Today, the order in which rooms are processed appears to be random, though to be honest I have not looked for any specific pattern. Spammy or destructive users may notice their bans propagating through smaller channels and make the larger channels their target where they will have a larger impact.

**Describe the solution you'd like**

I propose keeping a rough idea of the size of each room, and traversing the rooms from largest to smallest. The relative size of the rooms doesn't change rapidly over time, so stale information is not likely to be troublesome. Furthermore, starting with the largest rooms makes for a higher likelihood that the bans are effective near the start of the run.

**Describe alternatives you've considered**

None, really.

* I've wondered about making multiple Mjolnirs, but the cost/benefit is not there at all.
* I've wondered about adding more parallelism, but that is a big ball of wax.
* I've considered just leaving it the way it is: is it really a big deal if the bans take a couple extra minutes? Not really, but it would be nice if it were faster.

**Additional context**

I don't think it matters but my mjolnir is a user of an EMS-hosted Matrix server.

**Comment by Yoric** [comment]
@Gnuxie Since you're looking at banlists atm, how difficult do you think that would be?

**Comment by Gnuxie** [comment]
@Yoric I do not think that changing the order is the right way to solve the problem, if this is a real problem with large rooms we should attempt to manage it directly. Then we need to know why it is taking so long to apply bans to rooms large rooms. I suspect it is because we are requesting either the joined_members or the entire room state (if `fasterMembershipChecks` is not set to true). We can't really know though without testing and profiling this ourselves. 

@grahamc do you know if you have `fasterMembershipChecks` set to true in your configuration?

**Comment by grahamc** [comment]
I don't have fasterMembershipChecks set, shall I give that a shot?

**Comment by grahamc** [comment]
Note the docs on that setting say: 

> if your server struggles with /state requests then set this to true

and I don't know if my server struggles with them: it is hosted by EMS. That is why I haven't changed it :).
